UaClient/ServiceModel/Ua/EncodingDictionary.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+
+namespace Workstation.ServiceModel.Ua
+{
+    public class EncodingDictionary
+    {
+        private readonly Dictionary<NodeId, Type> encodingIdToTypeDictionary = new Dictionary<NodeId, Type>();
+        private readonly Dictionary<Type, NodeId> typeToEncodingIdDictionary = new Dictionary<Type, NodeId>();
+
+        public EncodingDictionary(IEnumerable<(ExpandedNodeId,Type)> table, IList<string> namespaceUris)
+        {
+            foreach (var (nodeId, type) in table)
+            {
+                ushort ns = nodeId.NodeId.NamespaceIndex;
+                var nsu = nodeId.NamespaceUri;
+                if (!string.IsNullOrEmpty(nsu))
+                {
+                    var i = namespaceUris.IndexOf(nsu!);
+                    if (i == -1)
+                    {
+                        continue;
```

Comment:
I intentionally do not use `ExpandableNodeId.ToNodeId` because it could silently convert a node with unknown namespace (not in the URI list) to a `ns=0` node, and in worst case overrride a default type.